File: oop/heritage/execice 3/src/Livre.java
package EX3;

public class Livre extends Document{
	private String auteur;
	private int nbPages;
	public Livre() {
		super();
		this.auteur="";
		this.nbPages=0;
	}
	public Livre(int num, String titre, String auteur, int nbPages) {
		super(num, titre);
		this.auteur=auteur;
		this.nbPages=nbPages;
	}
	public String getAuteur() {
		return auteur;
	}
	public void setAuteur(String auteur) {
		this.auteur = auteur;
	}
	public int getNbPages() {
		return nbPages;
	}
	public void setNbPages(int nbPages) {
		this.nbPages = nbPages;
	}
	String x = super.toString()	;@Override
	public String toString() {
		return "Livre [auteur=" + auteur + ", nbPages=" + nbPages + ", getAuteur()=" + getAuteur() + ", getNbPages()="
				+ getNbPages() + ", getNumEnregistrement()=" + getNumEnregistrement() + ", getTitre()=" + getTitre()
				+ ", toString()=" + x + ", getClass()=" + getClass() + ", hashCode()=" + hashCode()
				+ "]";
	}
	
	
}

File: oop/heritage/CPP/exercice3/src/Livre.cpp
#include "Livre.hpp"

Livre::Livre() : Document(), auteur(""), nbPages(0) {}

Livre::Livre(int num, const std::string& titre, const std::string& auteur, int nbPages)
    : Document(num, titre), auteur(auteur), nbPages(nbPages) {}

const std::string& Livre::getAuteur() const {
    return auteur;
}

void Livre::setAuteur(const std::string& newAuteur) {
    auteur = newAuteur;
}

int Livre::getNbPages() const {
    return nbPages;
}

void Livre::setNbPages(int newNbPages) {
    nbPages = newNbPages;
}

std::string Livre::toString() const {
    return "Livre [auteur=" + auteur + ", nbPages=" + std::to_string(nbPages) + ", base=" + Document::toString() + "]";
}
